packages/components/src/test/instance-filter-builder/Utils.test.ts

```diff
@@ -171,36 +171,63 @@ describe("filterRuleValidator", () => {
     ).to.be.eq("instance-filter-builder.error-messages.not-a-number");
   });
 
-  it("returns error message for invalid quantity rule", () => {
+  it("returns error message for numeric value set as an array of non-numeric values", () => {
     expect(
       filterRuleValidator({
         id: "test-id",
         groupId: "test-group-id",
-        property: quantityProperty,
+        property: numericProperty,
         operator: PropertyFilterRuleOperator.IsEqual,
         value: {
           valueFormat: PropertyValueFormat.Primitive,
-          value: undefined,
-          displayValue: "Invalid",
+          value: "[[10], [{}]]",
+          displayValue: "[[10], [{}]]",
         },
       }),
-    ).to.be.eq("instance-filter-builder.error-messages.invalid");
+    ).to.be.eq("instance-filter-builder.error-messages.not-a-number");
   });
 
-  it("does not return error message for valid numeric rule", () => {
+  it("returns error message for invalid quantity rule", () => {
     expect(
       filterRuleValidator({
         id: "test-id",
         groupId: "test-group-id",
-        property: numericProperty,
+        property: quantityProperty,
         operator: PropertyFilterRuleOperator.IsEqual,
         value: {
           valueFormat: PropertyValueFormat.Primitive,
-          value: 10,
-          displayValue: "10",
+          value: undefined,
+          displayValue: "Invalid",
         },
       }),
-    ).to.be.undefined;
+    ).to.be.eq("instance-filter-builder.error-messages.invalid");
+  });
+
+  [
+    { val: 10, testCase: "number" },
+    { val: 10.003, testCase: "float" },
+    { val: [10], testCase: "number array" },
+    { val: [[10]], testCase: "nested number array" },
+    { val: "[10]", testCase: "serialized array of numbers" },
+    { val: "{10}", testCase: "serialized struct" },
+    { val: "[10.003]", testCase: "serialized array of float numbers" },
+    { val: "[[10], [10]]", testCase: "serialized array of arrays of numbers" },
```

Comment:
I don't think we need all these cases. We need ones that check if numbers are handled correctly and one that handles serialize grouped values produces by `UniquValuesSelector`.